hub/core/version_control/commit_diff.py

```diff
@@ -55,6 +57,8 @@ def frombuffer(cls, data: bytes) -> "CommitDiff":
             int.from_bytes(data[27 + i * 8 : 35 + i * 8], "big")
             for i in range(num_updates)
         }
+        pos = 35 + num_updates * 8
+        commit_diff.cleared = bool(int.from_bytes(data[pos : pos + 1], "big"))
```

Comment:
this will not work for older datasets, should have an if-else condition here.

Reply:
also, preferably have this boolean flag stored at the start along with the other booleans

Reply:
this is fine right now since we are taking a slice of `data`. If we really want to move it up, what exactly will the if condition look like?